File: src/main/java/me/sd_master92/customvoting/commands/VotesCommand.java
package me.sd_master92.customvoting.commands;

import me.sd_master92.customfile.PlayerFile;
import me.sd_master92.customvoting.Main;
import me.sd_master92.customvoting.VoteFile;
import me.sd_master92.customvoting.constants.Messages;
import me.sd_master92.plugin.command.SimpleCommand;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.HashMap;

public class VotesCommand extends SimpleCommand
{
    private final Main plugin;

    public VotesCommand(Main plugin)
    {
        super(plugin, "votes");
        this.plugin = plugin;
    }

    @Override
    public void onCommand(CommandSender sender, String[] args)
    {
        if (args.length == 0)
        {
            if (sender instanceof Player)
            {
                Player player = (Player) sender;
                HashMap<String, String> placeholders = new HashMap<>();
                placeholders.put("%VOTES%", "" + new VoteFile(player, plugin).getVotes());
                placeholders.put("%s%", new VoteFile(player, plugin).getVotes() == 1 ? "" : "s");
                player.sendMessage(Messages.VOTES_COMMAND_SELF.getMessage(plugin, placeholders));
            }
        } else
        {
            String name = args[0];
            PlayerFile playerFile = PlayerFile.getByName(name, plugin);
            if (playerFile != null)
            {
                VoteFile voteFile = new VoteFile(playerFile.getUuid(), plugin);
                HashMap<String, String> placeholders = new HashMap<>();
                placeholders.put("%PLAYER%", "" + voteFile.getName());
                placeholders.put("%VOTES%", "" + voteFile.getVotes());
                placeholders.put("%s%", voteFile.getVotes() == 1 ? "" : "s");
                sender.sendMessage(Messages.VOTES_COMMAND_OTHERS.getMessage(plugin, placeholders));
            } else
            {
                sender.sendMessage(Messages.VOTES_COMMAND_NOT_FOUND.getMessage(plugin));
            }
        }
    }

    @Override
    public void onCommand(Player player, String[] strings)
    {

    }
}

File: src/main/kotlin/me/sd_master92/customvoting/commands/VotesCommand.kt
package me.sd_master92.customvoting.commands

import me.sd_master92.core.command.SimpleCommand
import me.sd_master92.customvoting.CV
import me.sd_master92.customvoting.constants.enumerations.Message
import me.sd_master92.customvoting.constants.enumerations.Setting
import me.sd_master92.customvoting.constants.enumerations.SoundType
import me.sd_master92.customvoting.constants.interfaces.Voter
import me.sd_master92.customvoting.getVotesPlaceholders
import me.sd_master92.customvoting.gui.pages.menus.VoteInfoMenu
import me.sd_master92.customvoting.sendText
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class VotesCommand(private val plugin: CV) : SimpleCommand(plugin, "votes")
{
    override suspend fun onCommand(sender: CommandSender, args: Array<out String>)
    {
        if (args.isEmpty())
        {
            getVotes(sender)
        } else
        {
            val name = args[0]
            val topVoter = Voter.getByName(plugin, name)
            if (topVoter != null)
            {
                getVotes(sender, topVoter)
            } else
            {
                sender.sendText(plugin, Message.VOTES_COMMAND_NOT_FOUND)
            }
        }
    }

    private suspend fun getVotes(sender: CommandSender, voter: Voter? = null)
    {
        if (voter != null)
        {
            if (plugin.config.getBoolean(Setting.VOTE_INFO_INVENTORY.path) && sender is Player)
            {
                SoundType.OPEN.play(plugin, sender)
                VoteInfoMenu(plugin, voter, voter.getName()).open(sender)
            } else
            {
                sender.sendText(plugin, Message.VOTES_COMMAND_OTHERS, voter.getVotesPlaceholders(plugin))
            }
        } else if (sender is Player)
        {
            val self = Voter.get(plugin, sender)
            if (plugin.config.getBoolean(Setting.VOTE_INFO_INVENTORY.path))
            {
                SoundType.OPEN.play(plugin, sender)
                VoteInfoMenu(plugin, self).open(sender)
            } else
            {
                sender.sendText(plugin, Message.VOTES_COMMAND_SELF, self.getVotesPlaceholders(plugin))
            }
        }
    }

    override suspend fun onCommand(player: Player, args: Array<out String>)
    {
    }

    init
    {
        withNoPermMessage(Message.NO_PERMISSION.getMessage(plugin))
    }
}